qcengine/programs/gaussian.py

```diff
@@ -0,0 +1,269 @@
+"""
+Calls the GAUSSIAN executable
+"""
+
+import os
+import re
+import tempfile
+import warnings
+from collections import defaultdict
+from typing import Any, Dict, List, Optional, Tuple
+import cclib
+from cclib.method import Nuclear
+
+import numpy as np
+from qcelemental import constants
+from qcelemental.models import AtomicInput, AtomicResult, Molecule, Provenance
+from qcelemental.molparse import regex
+from qcelemental.util import parse_version, safe_version, which
+
+from qcengine.config import TaskConfig, get_config
+
+from ..exceptions import InputError, UnknownError
+from ..util import disk_files, execute, temporary_directory
+from .model import ProgramHarness
+
+class GaussianHarness(ProgramHarness):
+
+    _defaults = {
+        "name": "gaussian",
+        "scratch": True,
+        "thread_safe": False,
+        "thread_parallel": True,
+        "node_parallel": False,
+        "managed_memory": True,
+    }
+    version_cache: Dict[str, str] = {}
+
+    class Config(ProgramHarness.Config):
+        pass
+
+    def found(self, raise_error: bool = False) -> bool:
+        return which(
+            "g09",
+            return_bool=True,
+            raise_error=raise_error,
+            raise_msg="Please install Gaussian. Check it's in your PATH with `which g09`."
+        )
```

Comment:
you probably also want to check if `cclib` is available when using this harness. see the nwchem harness (that needs networkx) for a model of how to set up looking for both.